construct/core.py

```diff
@@ -1549,18 +1552,28 @@ def __init__(self, keyfunc, cases, default=NoDefault, includekey=False):
         self.cases = cases
         self.default = default
         self.includekey = includekey
-        self.flagbuildnone = all(sc.flagbuildnone for sc in cases.values())
-        self.flagembedded = all(sc.flagembedded for sc in cases.values())
+        self.flagbuildnone = all(sc.flagbuildnone for sc in cases.values() if sc is not Pass)
+        self.flagembedded = all(sc.flagembedded for sc in cases.values() if sc is not Pass)
```

Comment:
This is here to fix the `If` helper function, where the else clause is a `construct.Pass` and has irrelevant flags for embedded status of the `Switch`.

This behavior is tested with `# Dynamic building of embedded.` at line 435 in the `tests/all_tests.py`

This pull request provides the behavior that if an element of a switch is embedded then it is embedded, if it is not then it is not.  This is done in the parse and build calls of switch to be minimal impact to other infrastructure.  It effectively ignores these init settings and updates them accordingly during parse and build according to the context passed in.

I agree how flags are handled in `Switch` is a sticking point with the  `all` call.  I think that when building if the flags are dynamic they should be updated.  Kinda like this statement.

```python
try:
    sc.update_dynamic_flags(context)
except AttrbuteError:
    pass
```
Or preferably, we allow switches and other dynamic constructs to manage their own context manipulations removing the burden from `Struct`'s `build`.  This is probably more robust in the long run.
More work  :weary: but something along the lines of this.

```python
if has_attr('manage_context', sc):
    context = sc.manage_context(context)
elif sc.flagembedded:
    # etc...
```